mgmt/rest/v2/task.go

```diff
@@ -33,14 +33,74 @@ import (
 	"github.com/julienschmidt/httprouter"
 )
 
+// TasksResp returns a list of created tasks.
+//
+// swagger:response TasksResponse
+type TasksResp struct {
+	// in: body
+	Body struct {
+		Tasks Tasks `json:"tasks"`
+	}
+}
+
+// TaskResp returns a task.
+//
+// swagger:response TaskResponse
+type TaskResp struct {
+	// in: body
+	Task Task `json:"task"`
+}
+
+// RemoveTaskError returns removing a task error.
+//
+// swagger:response TaskErrorResponse
+type RemoveTaskError struct {
+	// in: body
+	Message string `json:"message"`
+}
+
 type TasksResponse struct {
 	Tasks Tasks `json:"tasks"`
 }
 
+// TaskParam define the API path task id.
+//
+// swagger:parameters getTask watchTask updateTaskState removeTask
+type TaskParam struct {
+	// in: path
+	// required: true
+	ID string `json:"id"`
+}
+
+// TaskPostParams defines task POST and PUT content.
+// swagger:parameters addTask
+type TaskPostParams struct {
+	// Create a task.
+	//
+	// in: formData
+	//
+	// required: true
+	Task string `json:"task"yaml:"task"`
+}
+
+// TaskPutParams defines a task state
+//
+// swagger:parameters updateTaskState
+type TaskPutParams struct {
+	// Update the state of a task
+	//
+	// in: formData
+	//
+	// required: true
+	Action string `json:"action"`
+}
+
+// Task represents Snap task definition.
 type Task struct {
-	ID                 string            `json:"id"`
-	Name               string            `json:"name"`
-	Deadline           string            `json:"deadline"`
+	Version            int               `json:"version,omitempty"`
```

Comment:
You are right - task has a version used to differentiate between versions of **the task manifest parser** (right now, there is only one version: 1). For others reviewers - you can find more about it [here](https://github.com/intelsdi-x/snap/blob/master/docs/TASKS.md#version)

I have just a concern why `Task` struct did not have `Version` field before, but I need to be more acquainted with the flow of task version to say sth more. 

@candysmurf - what do you think about moving `Version` after `ID` and `Name`? Why: task ID and its name identified the task, so a good pattern keeps those two in the beginning. This is only a small suggestion, so up to you.